### ISSUE TITLE
Max Recursion error when stepping through Pydantic with pycharm debugger.

### ISSUE THREAD

**Issue by peter-boers** [samuelcolvin/pydantic#701]
#  Bug

When stepping through a Pydantic Model in the Pycharm debugger to find out why a number of validations are not working, I've encountered a Max Recursion error. Below you see a simplified version of the model that also hits the max recursion error.

For bugs/questions:
* OS: MacOS 10.14.6 (18G84)
* Python and Pydantic info
```bash
Python 3.7.3 (default, Mar 27 2019, 16:54:48) 
[Clang 4.0.1 (tags/RELEASE_401/final)] :: Anaconda, Inc. on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> import sys; print(sys.version)
3.7.3 (default, Mar 27 2019, 16:54:48) 
[Clang 4.0.1 (tags/RELEASE_401/final)]
>>> import pydantic; print(pydantic.VERSION)
0.30.1
>>> import pydantic; print(pydantic.compiled)
False
>>> 
```
* Pycharm build:
```
PyCharm 2019.1.3 (Professional Edition)
Build #PY-191.7479.30, built on May 29, 2019
JRE: 11.0.2+9-b159.60 x86_64
JVM: OpenJDK 64-Bit Server VM by JetBrains s.r.o
macOS 10.14.6
```

**Code Snippet**
```py
from pydantic import BaseModel

class TestModel(BaseModel):
    str_uuid: int

print(TestModel(str_uuid="string"))
```

**Stack Trace**
```bash
Connected to pydev debugger (build 191.7479.30)
RecursionError: maximum recursion depth exceeded while calling a Python object
Exception ignored in: '_pydevd_frame_eval.pydevd_frame_evaluator_darwin_37_64.get_bytecode_while_frame_eval'
RecursionError: maximum recursion depth exceeded while calling a Python object
Fatal Python error: Cannot recover from stack overflow.

Current thread 0x0000700009776000 (most recent call first):
  File "/Users/pboers/anaconda3/envs/workflows/lib/python3.7/threading.py", line 1222 in current_thread
  File "/Users/pboers/anaconda3/envs/workflows/lib/python3.7/site-packages/pydantic/main.py", line 278 in __getattr__
  File "/Users/pboers/anaconda3/envs/workflows/lib/python3.7/site-packages/pydantic/main.py", line 278 in __getattr__
  File "/Users/pboers/anaconda3/envs/workflows/lib/python3.7/site-packages/pydantic/main.py", line 278 in __getattr__
  File "/Users/pboers/anaconda3/envs/workflows/lib/python3.7/site-packages/pydantic/main.py", line 278 in __getattr__
  File "/Users/pboers/anaconda3/envs/workflows/lib/python3.7/site-packages/pydantic/main.py", line 278 in __getattr__
```

**Comment by samuelcolvin** [comment]
Best to create a ticket on pycharm and link to it here, hopefully they can be more helpful.

If there's something to do in pydantic to help this I'll happily look into it.

**Comment by pfrederiks** [comment]
The max recursion will not occur while stepping through the test in plain pdb. But a call to self within the `__init__` will cause a max recursion. Example:

```
> /Users/paul/src/pydantic/tests/test_infinite_recursion.py(1)<module>()
-> from pydantic import BaseModel
(Pdb) n
> /Users/paul/src/pydantic/tests/test_infinite_recursion.py(4)<module>()
-> class TestModel(BaseModel):
(Pdb) n
> /Users/paul/src/pydantic/tests/test_infinite_recursion.py(8)<module>()
-> print(TestModel(str_uuid="string"))
(Pdb) s
--Call--
> /Users/paul/anaconda3/envs/pydantic37/lib/python3.7/site-packages/pydantic/main.py(242)__init__()
-> def __init__(self, **data: Any) -> None:
(Pdb) self
*** RecursionError: maximum recursion depth exceeded while calling a Python object
(Pdb)
```
Our current theory is that this occurs when attributes are read from the Basemodel by any smart debugger before __values__ is set. The self.__getattr__ methods calls self.__values__ which if it is not present will in turn call self.__getattr__ to get self.__values__ etc.

**Comment by samuelcolvin** [comment]
(useful if you can use backticks when referencing code so we see `self.__getattr__` not self.__getattr__)

That makes sense.

Perhaps we could do a `hasattr()` check in `__getattr__`?

**Comment by samuelcolvin** [comment]
fixed by #712.
